pkg/apis/core/validation/validation.go

```diff
@@ -1433,6 +1435,14 @@ func validateCSIPersistentVolumeSource(csi *core.CSIPersistentVolumeSource, fldP
 		allErrs = append(allErrs, field.Required(fldPath.Child("driver"), ""))
 	}
 
+	if len(csi.Driver) > 63 {
+		allErrs = append(allErrs, field.TooLong(fldPath.Child("driver"), csi.Driver, 63))
+	}
+
+	if !csiDriverNameRexp.MatchString(csi.Driver) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("driver"), csi.Driver, fmt.Sprintf("Driver name must match : %s .For example, csi-hostpath", csiDriverNameRexp)))
```

Comment:
use `RegexError` to build a nicer error message (see example uses like `RegexError(configMapKeyErrMsg, configMapKeyFmt, "key.name", "KEY_NAME", "key-name")`)

Reply:
fixed